src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -1479,6 +1596,51 @@ void MPS::initialize_from_matrix(uint_t num_qubits, const cmatrix_t mat) {
   num_qubits_++;
 }
  
+//--------------------------------------------------
+// Algorithm for initialize_component:
+// 1. Centralize 'qubits'
+// 2. Compute the norm of 'qubits'
+// 3. Normalize the values in 'statevector' to the norm computed in (3)
+// 4. Create a new MPS consisting of 'qubits'
+// 5. Initialize it to 'statevector'
+// 6. Cut out the old section of 'qubits' in the original MPS
+// 7. Stick the new section of 'qubits' in the original MPS
+//---------------------------------------------------
```

Comment:
This is not what partial initialization is supposed to do. I'd expect this to fail in tests where the initialized qubits are entangled to other qubits. Correct partial initialization is a random instruction, where the initialized qubits are measured before the initialization. This is why the random generator is passed.

Reply:
I changed the functionality, as you pointed out. This required quite a few changes in reset and related functions. I added tests to `reset` and to `initialize` as I noticed two types of tests were missing: tests where the qubits are entangled before the `reset` or `initialize` and tests where the qubits may have moved in mps simulation.

Reply:
I think the changes require another review - @yaelbh , @vvilpas , @chriseclectic .